### PR TITLE
Sign Up

### PR DESCRIPTION
**Pull request by hazwanizafirah**

### CODE REVIEW

**Comment by hazwanizafirah** [comment]
ready

**Comment by hazwanizafirah** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+
+    if @user.nil?
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb:5]:
```suggestion
   unless@user
```

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-    <%= link_to t(:h_title), "#", id: "logo" %>
+    <%= link_to t(:h_title) , "#", id: "logo" %>
     <nav>
       <ul class="nav navbar-nav navbar-right">
-        <li><%= link_to t(:home), "#" %></li>
-        <li><%= link_to t(:help), "#" %></li>
-        <li><%= link_to t(:log_in), "#" %></li>
-        <li><%= link_to t(:c_en), locale: :en %></li>
-        <li><%= link_to t(:c_ja), locale: :jp %></li>
+      <li><%= link_to t(:home), home_path %></li>
```
Comment (ngocvt-0484) [inline app/views/layouts/_header.html.erb:6]: wrong indent

File: app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t "the_form_contains" %>
+      <%= pluralize @user.errors.count, "error" %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (ngocvt-0484) [inline app/views/shared/_error_messages.html.erb:8]: wrong indent

**Comment by hazwanizafirah** [comment]
ready

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+
+    unless@user
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:5]:
```suggestion
    unless @user
```

File: app/models/user.rb [hunk 5]
```diff
@@ -6,4 +6,11 @@ class User < ApplicationRecord
   format: {with: VALID_EMAIL_REGEX}
 
   has_secure_password
+  before_save :downcase_email
```
Comment (huongnt-2545) [inline app/models/user.rb:9]: this line is the handle same with line 3,
please remove line 3

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user, url: signup_path do |f| %>
+      <% if @user.errors.any? %>
+        <div id="error_explanation">
+          <div class="alert alert-danger">
+            <%= t (:the_form_contains)%>
+            <%= pluralize @user.errors.count, t(:error) %>.
+          </div>
+          <ul>
+          <% @user.errors.full_messages.each do |msg| %>
```
Comment (huongnt-2545) [inline app/views/users/new.html.erb:13]: wrong indent

Comment (hazwanizafirah) [reply app/views/users/new.html.erb:13]: Hi Miss. Actually I cannot find which one is the wrong indent. I've tried refer my sample app but it looks similar.

**Comment by hazwanizafirah** [comment]
ready

File: app/models/user.rb [hunk 7]
```diff
@@ -6,4 +6,10 @@ class User < ApplicationRecord
   format: {with: VALID_EMAIL_REGEX}
 
   has_secure_password
+
+  private
+
+  def downcase_email
```
Comment (huongnt-2545) [inline app/models/user.rb:12]: remove this method if it's not useful

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      flash[:danger] = t("user_not_found")
+      redirect_to home_path
+    end
+  end
+
+  def new
+     @user = User.new
+  end
+
+   def create
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:15]: wrong indent
```suggestion
  def create
```

File: app/views/users/new.html.erb (hunk 6 above)
Comment (huongnt-2545) [reply app/views/users/new.html.erb:13]:
```suggestion
            <% @user.errors.full_messages.each do |msg| %>
              <li><%= msg %></li>
            <% end %>
```

File: app/views/shared/_error_messages.html.erb (hunk 3 above)
Comment (huongnt-2545) [inline app/views/shared/_error_messages.html.erb:8]:
```suggestion
      <% @user.errors.full_messages.each do |msg| %>
        <li><%= msg %></li>
      <% end %>
```

**Comment by hazwanizafirah** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**